src/metabase/server/middleware/session.clj

```diff
@@ -68,6 +69,38 @@
                                                        metabase-embedded-session-cookie
                                                        metabase-session-timeout-cookie]))
 
+(def ^:private possible-session-cookie-samesite-values
+  #{:lax :none :strict nil})
+
+(defn- valid-session-cookie-samesite!
+  [value & [added-ex-data]]
+  (let [normalized-value (some-> value name str/lower-case keyword)]
+    (when-not (contains? possible-session-cookie-samesite-values normalized-value)
+      (throw (ex-info "Invalid value for session cookie samesite"
```

Comment:
On the question of translation: an error string should be wrapped with the `metabase.util.i18n.tru` macro if there's any chance of it being user-facing on the frontend. `tru` will translate the string into the current user's locale, while `trs` will translate it into the site-wide locale.

In this case though, I think it would only show up in the backend logs, so it's fine to leave it untranslated. (Most backend logs are actually translated into the site locale currently, but I think the plan is to move away from that—see [this Slack discussion](https://metaboat.slack.com/archives/CKZEMT1MJ/p1697553976266779).)

Reply:
And no need to add translations yourself; we use POEditor to source translations from volunteers prior to each major release.

Reply:
I'm not positive, but I would assume that `(throw (ex-info "foo" {:http-status 400}))` is going to bubble up to the frontend - so maybe I should actually wrap this error message in `tru`?

Or actually, maybe this means it's not a good idea to throw one error that might bubble up to either logs or an HTTP error - because we need to translate one but not the other. So I'll actually pull those apart.